gneiss/tests/test_sort.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+import pandas as pd
+import unittest
+from gneiss.sort import nichesort, mean_niche_estimator
+import pandas.util.testing as pdt
+
+
+class TestSort(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_mean_niche_estimator1(self):
+        gradient = pd.Series(
+            [1, 2, 3, 4, 5],
+            index=['s1', 's2', 's3', 's4', 's5'])
+        values = pd.Series(
+            [1, 1, 0, 0, 0],
+            index=['s1', 's2', 's3', 's4', 's5'])
+        m = mean_niche_estimator(values, gradient)
+        self.assertEqual(m, 1.5)
+
+    def test_mean_niche_estimator2(self):
+        gradient = pd.Series(
+            [1, 2, 3, 4, 5],
+            index=['s1', 's2', 's3', 's4', 's5'])
+        values = pd.Series(
+            [1, 3, 0, 0, 0],
+            index=['s1', 's2', 's3', 's4', 's5'])
+        m = mean_niche_estimator(values, gradient)
+        self.assertEqual(m, 1.75)
+
+    def test_mean_niche_estimator_bad_length(self):
+        gradient = pd.Series(
+            [1, 2, 3, 4, 5],
+            index=['s1', 's2', 's3', 's4', 's5'])
+        values = pd.Series(
+            [1, 3, 0, 0, 0, 0],
+            index=['s1', 's2', 's3', 's4', 's5', 's6'])
+
+        with self.assertRaises(ValueError):
+            mean_niche_estimator(values, gradient)
+
+    def test_mean_niche_estimator_missing(self):
+        gradient = pd.Series(
+            [1, 2, 3, 4, np.nan],
+            index=['s1', 's2', 's3', 's4', 's5'])
+        values = pd.Series(
+            [1, 3, 0, 0, 0],
+            index=['s1', 's2', 's3', 's4', 's5'])
+
+        with self.assertRaises(ValueError):
+            mean_niche_estimator(values, gradient)
+
+    def test_basic_nichesort(self):
+        table = pd.DataFrame(
+            [[1, 1, 0, 0, 0],
+             [0, 1, 1, 0, 0],
+             [0, 0, 1, 1, 0],
+             [0, 0, 0, 1, 1]],
+            columns=['s1', 's2', 's3', 's4', 's5'],
+            index=['o1', 'o2', 'o3', 'o4']).T
+        gradient = pd.Series(
+            [1, 2, 3, 4, 5],
+            index=['s1', 's2', 's3', 's4', 's5'])
+        res_table = nichesort(table, gradient)
+        pdt.assert_frame_equal(table, res_table)
+
+    def test_basic_nichesort_scrambled(self):
+
```

Comment:
Do you mind adding a short comment saying that columns 1 and 2 as well rows 1 and 2 are the ones that are in the wrong order? It took me some time to figure out what was scrambled :-)